src/adapter/src/coord/ddl.rs

```diff
@@ -651,13 +651,30 @@ impl Coordinator {
                 u64::MAX
             }),
         ));
-        let policies = self
+        let storage_policies = self
             .catalog()
             .entries()
-            .filter(|entry| entry.item().is_retained_metrics_relation())
+            .filter(|entry| {
+                entry.item().is_retained_metrics_relation() && entry.item().cluster_id().is_none()
+            })
             .map(|entry| (entry.id(), policy.clone()))
             .collect::<Vec<_>>();
-        self.update_storage_base_read_policies(policies)
+        let compute_policies = self
+            .catalog()
+            .entries()
+            .filter_map(|entry| {
+                if let (true, Some(cluster_id)) = (
+                    entry.item().is_retained_metrics_relation(),
+                    entry.item().cluster_id(),
+                ) {
+                    Some((cluster_id, entry.id(), policy.clone()))
+                } else {
+                    None
+                }
+            })
+            .collect::<Vec<_>>();
```

Comment:
Okay, I see now that it differs on `entry.item().cluster_id()` `None/Some` values. But that doesn't really partition by storage/compute objects because sources can have clusters associated (and MVs are also storage objects, though I'm not sure if that matters here).

Reply:
That's a good point, I suppose the logic is not fully correct for sources.

Perhaps I need to add a `fn is_compute_object(&self) -> bool` to `CatalogItem` which corresponds to which system the policies need to be set on.